
% Glutaredoxin domain-containing cysteine-rich protein 1, or GRXCR1, was the gene with the largest median decrease in posterior probability over the uniform model, with a median decrease of 0.08.


% SMAD3 has an average increase of 0.38, and median increase of 0.333

% NOTCH1 had an average increase of 0.33 and median increase of 0.32




% When reprioritizing genes it is equally important to take note the genes that have a decreased posterior probability of being driver genes.  The gene with the largest decrease in posterior probability across all analyses was Thrombospondin Type 1 Domain Containing 7A *THSD7A)  in HNSC.  THSD7A had a posterior under the uniform of 0.732, and a posterior under the functional prior of only 0.37.    THSD7A had a high log Bayes factor in HNSC of 5.65,  the highest log Bayes factor observed for that gene in any of the 18 cancer types for which data was available (the mean uniform posterior probability of THSD7A being a driver gene was 0.049).
% % THSD7A is not listed as a cancer gene in the intogen database, and 


%The gene with the lowest prior in all analyses is BIN1 in LUAD, with a prior probability of \num{1.79e-4}.  BIN1 is annotated with two GO terms with non-zero effect sizes in the LUAD multivariate model.  Both of these terms negative estimates: GO:0016032, ``viral process'', has an estimate of -1.72, and ``positive regulation of apoptotic process'', GO:0043065, has an estimate of -0.33.


The \texttt{driverMAPs} method identifies mutational driver genes by testing genes for an excess of somatic point mutations as compared to background mutational processes. Somatic SNV are an important class of somatic variants, (they are the most frequently occuring source of somatic variation) but somatic insertion deletions, structural variants and retrotranspositions also represent significant sources of somatic variation\cite{PANCAN}.  If a mutational driver gene does not rely on coding somatic SNVs to unlock their oncogenic potential, (e.g the gene fusion of BCR and ABL1 created by reciprocal translocation of chromosomes 9 and 22\cite{philly}) it will be missed by methods relying on counting somatic SNVs.